### ISSUE TITLE
Metrics for Service Binding States

### ISSUE THREAD

**Issue by MatthiasWinzeler** [cloudfoundry-incubator/service-fabrik-broker#1314]
**Is your feature request related to a problem? Please describe.**
Hi folks!
We were delighted to see that the interoperator already comes with metrics regarding cluster usage & service instance states (and a sample dashboard for it).

**Describe the solution you'd like**
However, for service bindings, there is currently no metrics. Having metrics `service_binding_state` similar to `service_instance_state` would be very nice (especially now that interoperator supports async bindings).

**Describe alternatives you've considered**
None

**Additional context**
Would you accept a PR for this? Happy to contribute!

**Comment by swati1102** [comment]
Hi @MatthiasWinzeler 
Thank you for reaching out and proposing to also raise PR for the same. We would in general be happy to take the PRs.
But, since we are in planning phase for Q2, we can may be add the above mentioned feature request in our scope and plan to work on it. Would that be fine with you?

**Comment by MatthiasWinzeler** [comment]
Hi @swati1102 

Sounds great! Q2 would be perfect for us. Thanks for your effort and let us know when we can support.

**Comment by Pooja-08** [comment]
@MatthiasWinzeler we have added service_binding_state metrics to show the binding_id and the id of the corresponding service_instance this binding belongs to. Here is a sample promql output:


```
interoperator_service_bindings_state{app_kubernetes_io_managed_by="Helm", binding_id="0abc2107-de86-408d-8617-800935b84028", control_plane="interoperator-multiclusterdeployer-controller-manager", instance="xx.xx.xx", instance_id="80ca2362b-6561-4673-ad24-111d7ac86cfd", job="kubernetes-service-endpoints", kubernetes_name="interoperator-multiclusterdeployer-metrics-service", kubernetes_namespace="interoperator", kubernetes_node="shoot--interop--pooja-dev-worker-j5mor-z1-6b645-jb924"} | 1
interoperator_service_bindings_state{app_kubernetes_io_managed_by="Helm", binding_id="0abc2107-de86-408d-8617-800935b84038", control_plane="interoperator-multiclusterdeployer-controller-manager", instance="xx.xx.xx", instance_id="80ca2362b-6561-4673-ad24-111d7ac86cfd", job="kubernetes-service-endpoints", kubernetes_name="interoperator-multiclusterdeployer-metrics-service", kubernetes_namespace="interoperator", kubernetes_node="shoot--interop--pooja-dev-worker-j5mor-z1-6b645-jb924"} | 1
interoperator_service_bindings_state{app_kubernetes_io_managed_by="Helm", binding_id="0ceb2107-de86-408d-8617-800935b84108", control_plane="interoperator-multiclusterdeployer-controller-manager", instance="xx.xx.xx", instance_id="65ca2362b-6561-4673-ad24-111d7ac86cfd", job="kubernetes-service-endpoints", kubernetes_name="interoperator-multiclusterdeployer-metrics-service", kubernetes_namespace="interoperator", kubernetes_node="shoot--interop--pooja-dev-worker-j5mor-z1-6b645-jb924"} | 0
interoperator_service_bindings_state{app_kubernetes_io_managed_by="Helm", binding_id="1abc2107-de86-408d-8617-800935b84038", control_plane="interoperator-multiclusterdeployer-controller-manager", instance="xx.xx.xx", instance_id="81ca2362b-6561-4673-ad24-111d7ac86cfd", job="kubernetes-service-endpoints", kubernetes_name="interoperator-multiclusterdeployer-metrics-service", kubernetes_namespace="interoperator", kubernetes_node="shoot--interop--pooja-dev-worker-j5mor-z1-6b645-jb924"} | 0

```

Also, attached are few sample dashboards that can be generated using the binding metrics:
**service_binding_state** : represents the state of all the bindings
**service_binding_count**: represents the total count of the bindings present in the cluster
**service_bindings_grouped_by_instances**: represents the bindings grouped by the service instances
**service_bindings_in_failed_state**: represents the failed bindings in the cluster

<img width="1789" alt="binding_state" src="https://user-images.githubusercontent.com/18735384/124735924-1fd0c080-df34-11eb-89bf-8d2b26ce5b9e.png">


Please let us know if this helps.

**Comment by MatthiasWinzeler** [comment]
@Pooja-08 sounds really cool! do you already have a plan when it should land in a release?

**Comment by Pooja-08** [comment]
@MatthiasWinzeler tentatively by July end.

**Comment by MatthiasWinzeler** [comment]
@Pooja-08 cool, sounds great!

**Comment by pivotal-marcela-campo** [comment]
Closing this feature request as it has been released in [v0.17.0](https://github.com/cloudfoundry-incubator/service-fabrik-broker/releases/tag/v0.17.0)
